internal/util/projutil/exec.go

```diff
@@ -0,0 +1,116 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package projutil
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	homedir "github.com/mitchellh/go-homedir"
+)
+
+func ExecCmd(cmd *exec.Cmd) error {
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	if err := cmd.Run(); err != nil {
+		return fmt.Errorf("failed to exec %#v: %v", cmd.Args, err)
+	}
+	return nil
+}
+
+// GoBuildOptions configures "go build" and "go test".
+type GoBuildOptions struct {
+	// BinName is the name of the compiled binary, passed to -o.
+	BinName string
+	// BuildArgs are args passed to "go build", aside from "-o {bin_name}".
+	BuildArgs []string
+	// BuildPath is the path containing main package file.
+	BuildPath string
+	// Env is a list of environment variables to pass to "go build";
+	// exec.Command.Env is set to this value.
+	Env []string
+	// Dir is the dir to run "go build" in; exec.Command.Dir is set to this value.
+	Dir string
+	// NoGoMod determines whether to set the "-mod=vendor" flag.
+	// If false, GoBuild will use go modules.
+	// If true, "go build" will not use modules.
+	NoGoMod bool
+}
+
+const (
+	goBuildCmd = "build"
+	goTestCmd  = "test"
+)
+
+// GoBuild runs "go build" configured with opts.
+func GoBuild(opts GoBuildOptions) error {
+	return goCmd(goBuildCmd, opts)
+}
+
+// GoBuild runs "go test" configured with opts.
+func GoTest(opts GoBuildOptions) error {
+	return goCmd(goTestCmd, opts)
+}
+
+// goCmd runs "go {cmd}", where cmd is either "build" or "test".
+func goCmd(cmd string, opts GoBuildOptions) error {
+	bargs := []string{cmd}
+	if opts.BinName != "" {
+		bargs = append(bargs, "-o", opts.BinName)
+	}
+	if cmd == goTestCmd {
+		bargs = append(bargs, opts.BuildPath)
+	}
+	bargs = append(bargs, opts.BuildArgs...)
+	// Modules can be used if either GO111MODULE=on or we're not in $GOPATH/src.
+	if !opts.NoGoMod {
```

Comment:
Is it okay to include this block before we've introduced the other go mod changes? Since we don't set `opts.NoGoMod` to `true`, this block will execute.

Reply:
Changed to `opts.GoMod` for now. I'll modify this field once #1001 is merged.